lib.rs

```diff
@@ -87,16 +140,24 @@ impl<D: Decorator> Format<D> {
         let r_decorator = self.decorator.decorate(record);
 
 
-        try!(self.print_msg_header(io, &r_decorator, record));
+        let mut skip_msg_comma = try!(self.print_msg_header(io, &r_decorator, record));
         let mut serializer = Serializer::new(io, r_decorator);
 
         for (k, v) in logger_values.iter() {
-            try!(serializer.print_comma());
+            if !skip_msg_comma {
+                try!(serializer.print_comma());
+            } else {
+                skip_msg_comma = false;
```

Comment:
And then this line should be `self.comma_need |= true`, to make it clear that this flag works through.... hmm.. saturating to `true` on the first key/value printed.